internal/parse/parse.go

```diff
@@ -19,22 +19,63 @@ const (
 
 // Suffixes for byte measurements that are valid to PostgreSQL
 const (
-	TB = "TB"    // terabyte
-	GB = "GB"    // gigabyte
-	MB = "MB"    // megabyte
-	KB = "kB"    // kilobyte
-	B  = ""      // no unit, therefore: bytes
+	TB = "TB" // terabyte
+	GB = "GB" // gigabyte
+	MB = "MB" // megabyte
+	KB = "kB" // kilobyte
+	B  = ""   // no unit, therefore: bytes
 )
 
+// Suffixes for time measurements that are valid to PostgreSQL
+// Valid time units are us (microseconds), ms (milliseconds), s (seconds), min (minutes), h (hours), and d (days).
+// https://www.postgresql.org/docs/current/config-setting.html#20.1.1.%20Parameter%20Names%20and%20Values
 const (
-	errIncorrectFormatFmt      = "incorrect PostgreSQL bytes format: '%s'"
-	errCouldNotParseBytesFmt   = "could not parse bytes number: %v"
-	errCouldNotParseVersionFmt = "unable to parse PG version string: %s"
-	errUnknownMajorVersionFmt  = "unknown major PG version: %s"
+	Microseconds = "us"
+	Milliseconds = "ms"
+	Seconds      = "s"
+	Minutes      = "min"
+	Hours        = "h"
+	Days         = "d"
+)
+
+func VerifyTimeUnits(units string) error {
```

Comment:
Good article. I totally agree with it.

This is an area where I feel that Go leaves a lot to be desired. The fact that there are no "real" enums makes this kind of thing so hacky. 

I created a type for `Profile` and a `GetProfile` function (which might be better named `ParseProfile`). However, just because I have a variable of type `Profile` in hand does not guarantee that it represents a valid value because someone could do `p := Profile(1982)` which is an invalid value but there's no way for the type system to prevent it.

<img width="763" alt="image" src="https://user-images.githubusercontent.com/4974668/186489708-d42506ad-5a4c-4698-8476-b627d16dac1f.png">

<img width="830" alt="image" src="https://user-images.githubusercontent.com/4974668/186489774-4c1f881b-3b73-4bf7-b624-35b8869ef79c.png">

Reply:
Thanks for the feedback. Went ahead and changed to enum-ish types. Feels better than strings. Still wish for "real" enums.

4398a5a807c1005878e4157fd8a806a021c41343
a391d0c11901177d84775ff2f613fe6276f86e2b